tools/wavetool.py

```diff
@@ -0,0 +1,157 @@
+        description='A Tool to Generate and Manipulate Wave Files.')
+    parser.add_argument('-v', '--version', action='version', version='%(prog)s 1.0')
+    parser.add_argument('-g', '--generate', type=str, help='generate specified types of wave\n'
+    'parameter format: "wave_type; amplitude(0.0~1.0); frequency; phase; sample rate; channel; format; duration"\n'
+    'Note: please use float point precision\n'
+    'Supported wave type: sine, cosine\n'
+    'Support format: S8, S16, S32, F32\n')
+    parser.add_argument('-o', '--output', type=str, help='Path to store generated files', default='.')
+    parser.add_argument('-p', '--preset', type=str, help='Commonly used wave presets\n'
+    'Supported presets are: sine_1k_16bit_48k_10s')
+    parser.add_argument('-c', '--compare', type=str, help='compare recorded wave and reference wave\n'
+    'parameter format: "mode;recorded_wave_path; reference_wave_path"\n'
+    'Supported comparison modes are: HWEchoRef')
+    return vars(parser.parse_args())
+
+def parse_params_and_gen_wav(cmd_param):
+    supported_type = ['sine', 'cosine']
+    param_keys = ['type', 'amp', 'freq', 'phase', 'sample_rate', 'channel', 'format', 'duration']
+    split_cmd_params = cmd_param.split(';')
+    wave_params = [param.strip() for param in split_cmd_params]
+    param_dict = dict(zip(param_keys, wave_params))
+    if param_dict['type'] not in supported_type:
+        print("Unsupported wave type")
+        sys.exit(1)
+    if param_dict['type'] in ['sine', 'cosine']:
+        wave_data = generateSinusoid(**param_dict)
+        return wave_data, param_dict
+
+def do_wave_compare(cmd_args):
+    supported_mode = ['HWEchoRef']
+    cmd_line = cmd_args['compare'].strip().split(';')
+    comparison_mode = cmd_line[0].strip()
+    if comparison_mode not in supported_mode:
+        print("Unsupported comparison mode: %s" % comparison_mode)
+        print('Supported comparison modes are: ' + str(supported_mode))
+        sys.exit(1)
+    wav_path, ref_path = [path.strip() for path in cmd_line[1:]]
+    if not os.path.exists(wav_path):
+        print('Recorded wave path does not exist: %s' % wav_path)
+        sys.exit(1)
+    if not os.path.exists(ref_path):
+        print('Reference wave path does not exist: %s' % ref_path)
+        sys.exit(1)
+    if comparison_mode == 'HWEchoRef':
+        compare_wav_hw_echo_ref(wav_path, ref_path)
+
+def save_wave(wave_data, path, **wave_params):
+    supported_format = ['S8', 'S16', 'S32', 'F32']
+    wave_path = path
+    if path == '.' or not path.endswith('wav'):
+        wave_name = wave_params['type'] + '_' + wave_params['channel'] + 'ch_' + wave_params['freq'] + 'Hz_' \
+        + wave_params['sample_rate'] + '_' + wave_params['format'] + '.wav'
+        wave_path = path + '/'  + wave_name
+    wave_format = wave_params['format'].upper()
+    if wave_format not in supported_format:
+        print('Unsupported wave format: %s' % wave_format)
+        sys.exit(1)
```

Comment:
https://docs.python.org/3/library/argparse.html#choices does this check for free.